### DISCUSSION TITLE
MINTEMP Error with self compiled firmware

### DISCUSSION THREAD

**Discussion by Daires1** [mriscoc/Ender3V2S1#1022] [Q&A] [answered]
Hello guys!

I hope you can help me.
I tried to updated the Firmware of my near stock Ender 3V2 from the old Jyers firmware. 
Because I use different fans, I have to kick start them with the FAN_KICKSTART_TIME command within Configuration_adv.h. So I wanted to self compile.

Here is what I have done:
- Loaded the source code from the latest august release and from the Special_Configuration repository. 
- Used the Configurator (Ender3V2, 422, MM, DWIN, T1, MPC) and copied the Configuration.h/Configuration_adv.h in the Marlin folder, the platformio.ini in the root folder.
- Enabled FAN_KICKSTART_TIME and FAN_KICKSTART_POWER within the Configuration_adv.h file.
- Used Auto Build Marlin with environment STM32F103RE_creality (512k)
- Copied the resulting .bin on fresh formated SD-Card and flashed it on the printer.
- Printer is showing the right firmwarename with "Costume firmware" at the beginning.

The problem now is, everytime I try to heat up the HotEnd I get "Temperture to low" and then the MINTEMP error.

I checked the board and chip number twice: Board is v4.2.2, chip is STM32F103RE6T
I tried the precompiled august firmware  Ender3V2-422-MM-MPC-20230812.bin and everything works just fine.
I tried to compile the old Jyers firmware with the environment STM32F103RE6T_creality (512k) again and the HotEnd works fine.

There seems to be something I´m missing but I can´t find the solution. I checked the Configuration.h/Configuration_adv.h file for anything MINTEMP relevant and out of the ordinary but couldn´t find anything what could cause the problem. I tried to use PID but got the same error. I tried for two days now to get this to work but everytime I self compile the august release the MINTEMP error shows.

So, if anyone has an idear what I´m doing wrong, it would be kind if you could point me in the right direction.

Greetings!

**Comment by mriscoc** [comment]
Please try recently published sources.

**Reply by Daires1** [reply]
Hi mrisoc!
First, thank you for your reply!
I have tried the new repository you commited 2hours ago from both, the [Ender3V2S1](https://github.com/mriscoc/Ender3V2S1) and [Special_Configurations](https://github.com/mriscoc/Special_Configurations).

Used the Configurator.pyw to generate the config files like described above and compiled it again with Auto Build Marlin with environment STM32F103RE_creality (512k). The MINTEMP error still remains.

BUT, I don´t know why and I actually can´t reproduce it, at some point I was able to heat up the Hotend without the error over the Control - Temperature - Hotend menue. To check things, I rebooted the printer and tried again but the MINTEMP error happend again.
Here is what I tried to see if something was causing the error:

- Heating up over the shortcuts directly after booting -> MINTEMP error
- Heating up over the Control - Temperature - Hotend menue directly after booting -> MINTEMP error
- Heating up over the Control - Temperature - Hotend menue after enabling bed heating and Partcooling fan -> MINTEMP error
- Heating up over the Control - Temperature - Hotend menue after waiting 5min -> MINTEMP error

At this point, I would have said there is something is wrong with the hardware.
To test this, I got the precompiled release Ender3V2-422-MM-MPC-20230812.bin again and flashed it. To my suprise I got no MINTEMP error at any time (tried the same steps as described above).
Also the reading for the Hotend temperature is always the ambient temperature with both firmwares.

I have read the discussion Pinter Killed. Err: MINTEMP. #821 who seemed to had the same problem, but could not actually solve it.

**Reply by mriscoc** [reply]
## Tests:

Using the default configuration files (Ender3V2-422-MM) included in the main repository and enable:
```cpp
#define FAN_KICKSTART_TIME  100  // (ms)
#define FAN_KICKSTART_POWER 180  // 64-255
```
- Build with 512K environment (RET6 SoC),
- Flash firmware
- Restore defaults
- Store settings
- Set control / temperature / Hotend to 185 °C

All is working as expected.

**Reply by Daires1** [reply]
Thank you again for your answer!
I tried your suggestion, but it didn´t work. That leaved me to believe that something on my files or on my way to compile was wrong.
Long story short: I deinstalled Visual Studio Code, which was the user edition, and installed the system edition with administrator privilege. Afterwards it worked like a charm first try.

I want to thank you for your time and help and for your great effort on this firmware!
And shame on me for stealing your time.
Greetings!

**Reply by Daires1** [reply]
Error is still happening.
Tried to start a print today and got the MINTEMP error again. Measured the thermistor, 89,7 kOhm at around 27 °C, so looking good. Will try some other hardware stuff and will come back here later.

**Reply by Daires1** [reply, marked as answer]
So, after a lot of try and error, I´m back at a working printer.

I checked the hardware and everything was fine, so i was back to the software again.

To make sure I´m not crazy, I flashed the original precompiled firmware from August Ender3V2-422-MM-MPC-20230812.bin again and checked if really everything was working fine here. I got no MINTEMP error with this. 
Afterwards I compiled it myself and flashed the selfcompiled firmware. This lead to the MINTEMP error. Again.

Following an intuition my next step was to take the source of the May release [Ender-3 V2/S1 Professional Firmware 20230522](https://github.com/mriscoc/Ender3V2S1/releases/tag/20230522) and use the configuration files from the Ender3V2-422-MM folder to compile a firmware. To my supprise, I did not get the MINTEMP error with it. So I compared the configuration files from both releases.

The first thing I notices was the use of "ANY" instead of "EITHER" as comparison and "ALL" instead of "BOTH". I´m not so familiar with the different commads but i do not think, that this is the culprit. Of course there are other tweaks within the configuration files.

In any case, I changed the config files from the May release to use MPC, FAN_KICKSTART_TIME and FAN_KICKSTART_POWER and flashed it afterwards. This selfcompiled firmware gave me no MINTEMP errors.

To double check I compiled another August release, but the MINTEMP error occurt directly.

So right now I´m back at the May release with my changes. But for the love of my life, I can´t find the reason why a selfcompiled August release does not work. I think there is a difference somewhere I´m not looking, but I´m not so knowledgeable in Marlin and firmwares in generale.
If someone with a better knowledge of Marlin and everything behind this firmware has an insight, I would be courious to know.

For now, I stay with the May release and hope this piece of information can help someone or to find the problem I´m not able to find.

**Comment by tokolist** [comment]
@Daires1 please see my solution here https://github.com/mriscoc/Ender3V2S1/discussions/1092#discussioncomment-7101947
